### DISCUSSION TITLE
list of events that an element can subscribe to

### DISCUSSION THREAD

**Discussion by nathanjshaffer** [zauberzeug/nicegui#4143] [Q&A]
### Question

In the documetation the an alement can subscribe to an event via on():
```
on(type: str, handler: Optional[events.Handler[events.GenericEventArguments]] = None, args: Union[None, Sequence[str], Sequence[Optional[Sequence[str]]]] = None, throttle: float = 0.0, leading_events: bool = True, trailing_events: bool = True, js_handler: Optional[str] = None) -> Self

Subscribe to an event.

    param type:	  name of the event (e.g. "click", "mousedown", or "update:model-value")
    param handler:	callback that is called upon occurrence of the event
    param args:	arguments included in the event message sent to the event handler (default: None meaning all)
    param throttle:	minimum time (in seconds) between event occurrences (default: 0.0)
    param leading_events:
 	    whether to trigger the event handler immediately upon the first event occurrence (default: True)
    param trailing_events:
 	    whether to trigger the event handler after the last event occurrence (default: True)
    param js_handler:
 	    JavaScript code that is executed upon occurrence of the event, e.g. (evt) => alert(evt) (default: None)
```

the param type section lists a couple examples `(e.g. "click", "mousedown", or "update:model-value")`, but I can't find a full list of what events are available via the event type parameter.  The Event & Actions section of the documentation seems to really only cover app wide events, rather than events that are specific to an element.

**Comment by astariul** [comment]
Here is the list I'm using : https://developer.mozilla.org/en-US/docs/Web/API/Element#events

**Comment by ed2050** [comment]
Short answer - there is no full list.  There are hundreds of events, some deprecated, some browser-specific.  Documentation is scattered.

You can find links to a partial list under [Generic Events](https://nicegui.io/documentation/section_action_events#generic_events) in nicegui docs.

Other good sources of event info:
- HTML standard 
    - [element events](https://html.spec.whatwg.org/multipage/indices.html#events-2)
    - [media events](https://html.spec.whatwg.org/multipage/media.html#mediaevents)
    - [drag and drop events](https://html.spec.whatwg.org/multipage/dnd.html#dndevents)
- MDN [event overview](https://developer.mozilla.org/en-US/docs/Web/Events) listing some common events
- MDN [list of all API Event types](https://developer.mozilla.org/en-US/docs/Web/API/Event) - have to chase links several levels deep to find actual events based on each type.
- MDN [document events](https://developer.mozilla.org/en-US/docs/Web/API/Document) (see section Events)
